From 564a8b6e5affc790126513bf618a6240175b525f Mon Sep 17 00:00:00 2001
From: Andrew Liu <shengping.liu@windriver.com>
Date: Thu, 16 Dec 2010 14:43:06 +0800
Subject: [PATCH 15/28] Remove possible recursive locking in p4080ds_mdio_reset

In the p4080ds_mdio_reset, have hold mdio_lock, enter into
fsl_pq_mdio_reset(priv->real_bus->reset) still tries to acquire
mdio_lock, although, here private real_bus is not same to mdio_bus,
potentially exists recursive locking.

Signed-off-by: Andrew Liu <shengping.liu@windriver.com>
---
 arch/powerpc/platforms/85xx/corenet_ds_mdio.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/arch/powerpc/platforms/85xx/corenet_ds_mdio.c b/arch/powerpc/platforms/85xx/corenet_ds_mdio.c
index 5371438..7767f00 100644
--- a/arch/powerpc/platforms/85xx/corenet_ds_mdio.c
+++ b/arch/powerpc/platforms/85xx/corenet_ds_mdio.c
@@ -85,9 +85,7 @@ static int p4080ds_mdio_reset(struct mii_bus *bus)
 {
 	struct p4080ds_mdio *priv = bus->priv;
 
-	mutex_lock(&bus->mdio_lock);
 	priv->real_bus->reset(priv->real_bus);
-	mutex_unlock(&bus->mdio_lock);
 
 	return 0;
 }
-- 
1.6.5.2

